corehq/apps/app_manager/tests/test_suite_inline_search.py

```diff
@@ -559,12 +630,28 @@ def test_child_module_with_inline_search_entry(self):
             </command>
             <instance id="casedb" src="jr://instance/casedb"/>
             <instance id="commcaresession" src="jr://instance/session"/>
-            <instance id="results:inline" src="jr://instance/remote/results:inline"/>
+            <instance id="results:child_instance" src="jr://instance/remote/results:child_instance"/>
+            <instance id="results:parent_instance" src="jr://instance/remote/results:parent_instance"/>
             <session>
-              <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='case'][@status='open']"
+              <query default_search="false" dynamic_search="false" storage-instance="{self.m0.search_config.get_instance_name()}" template="case" url="http://localhost:8000/a/test_domain/phone/search/123/">
+                <title>
+                  <text>
+                    <locale id="case_search.m0.inputs"/>
+                  </text>
+                </title>
+                <data key="case_type" ref="'case'"/>
+                <prompt key="name">
+                  <display>
+                    <text>
+                      <locale id="search_property.m0.name"/>
+                    </text>
+                  </display>
+                </prompt>
+              </query>
+              <datum id="case_id" nodeset="instance('{self.m0.search_config.get_instance_name()}')/results/case[@case_type='case'][@status='open'][not(commcare_is_related_case=true())]"
```

Comment:
Ya the `case_id` datum is coming from the parent module due to `parent-select` (parent relationship). With "parent-select", isn't it supposed to present the search screen twice? The workflow would be :
Select Submodule Menu -> Case search parent case -> case search cases that are children of selected parent case -> open form

>  we should be able to get the case ID from the commcaresession instance instead of re-querying for it.

Are you talking about the case ID of the parent case we're interested in?  The `query` element populates the instance that the`<datum id="case_id"...>` nodeset references right? I'm not following how `case_id` would be populated in the `commcaresession` instance without the `query` element

Reply:
> Because this is a child module, the user would already have selected the parent case by the time we get to this point in the navigation so the case ID would already be in the session.

but the child case id still needs to be selected right ? 


>  Looking at how normal child modules work I see they still have the full nodeset expression. I

But they should filter on parent case selected I guess ? 


> I guess we'll just have to test this and see if it presents an extra search screen or not.

Would the child cases be in the search results instance though ? If yes,  I guess the search screen is not really useful and we can just write the nodeset for child datum so that it filters the results instance with parent case selected to only show the related cases.

Reply:
@snopoke Could you clarify some more, I'm still unsure how to proceed from here.

I looked into it, and this query is added because the parent module `auto_launch=True`, meaning it's always going to case search first. If i set `auto_launch=False`, then that `query` element is no longer present. And the `datum` you highlighted becomes this:
`<datum detail-select="m0_case_short" id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='case'][@status='open']" value="./@case_id"/>`
which gets the parent case from `casedb`.

Otherwise, by extra screen, do you mean a 3rd screen? When I test it on Web Apps, I see two search screens which is what I expect.
Select Parent Module -> Select Submodule Menu -> Case search parent case (Search Screen1)-> case search for cases that are children of selected parent case (SearchScreen2)-> select form
SearchScreen1 collects the data for datum `case_id` and Search Screen2 collects the data for datum `case_id_case`

Reply:
OK, that seems fine then. I think what Shubham was suggesting is a possible optimization but I'm not sure it would always work.